src/main/java/frc/robot/subsystems/Chamber.java

```diff
@@ -136,20 +138,37 @@ public boolean shootNote(double leftSpeed, double rightSpeed) {
     }
   }
 
+  public boolean isNoteAligned() {
+    if (isNoteDetected() && !m_noteUnderSensor) {
+      m_counter++;
+      m_noteUnderSensor = true;
+    } else if (!isNoteDetected()) {
+      m_noteUnderSensor = false;
+    }
+    if (m_counter == 2 && !m_noteUnderSensor) {
```

Comment:
The magic number 2 here is problematic. It's a configurable constant, so it belongs in `ChamberConstants`. I can't tell for sure what the intent is either. Is the idea to have some number of scheduler quanta pass as `m_noteUnderSensor` toggles back and forth? If so it makes assumptions about scheduler loop timing that aren't robust, because the scheduler loop could be reconfigured (that's why we use `kDt` rather than hard-coding 0.02), plus as you've seen, the scheduler can fall behind if something computationally expensive occurs. As written, this is "wait at least 40ms, but possibly much longer". Take a look at how the `CLEARING` case below uses time deltas. That approach isn't susceptible to additive error of multiple scheduler loop overruns.